### ISSUE TITLE
How do I design my loss fuction like smoothL1？

### ISSUE THREAD

**Issue by kelisiya** [CoinCheung/pytorch-loss#3]
I want to design a loss fuction when GT is [0,1] use L2 loss ;[0-1] use L1 loss ;
How to achieve it use pytorch？
Thanks

**Comment by CoinCheung** [comment]
I have not seen loss like that, maybe you can implement that directly with pytorch operators.

**Comment by kelisiya** [comment]
yes, I want to&nbsp; design a loss use pytorch , but I don't know how to deal when value is 0 or 1 and 0-1 ;&nbsp;I think it's like smoothL1,but I&nbsp; use pytorch unskilled.

**Comment by CoinCheung** [comment]
Maybe you can try `torch.where` to switch between the two losses according to the input conditions.

**Comment by kelisiya** [comment]
Thanks your reply，but how to del 0 or 1 ? torch don't have logit or ?&nbsp;

**Comment by CoinCheung** [comment]
what do you mean by "del 0 or 1" ?

**Comment by kelisiya** [comment]
Emmm，I want to use L1 loss in label=0 and label=1，use L2 loss in label =[0-1] like 0.5，i dont know how to use torch.where in label =0 and label =1



&nbsp;---Original---

**Comment by CoinCheung** [comment]
As far as I know, we do not need to consider single points when we compute loss in practice. You need continuous loss funtions to compute the gradient, it will not be meaningful if your loss function has singular points that cannot compute the graident.

**Comment by kelisiya** [comment]
《A Late Fusion CNN for Digital Matting》,http://openaccess.thecvf.com/content_CVPR_2019/papers/Zhang_A_Late_Fusion_CNN_for_Digital_Matting_CVPR_2019_paper.pdfthe&nbsp;formula <4&gt;&nbsp;
